charts/m4d-crd/templates/crds.yaml

```diff
@@ -2003,6 +1661,72 @@ status:
 ---
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.3.0
+  creationTimestamp: null
+  name: m4dstorageaccounts.app.m4d.ibm.com
+spec:
+  group: app.m4d.ibm.com
+  names:
+    kind: M4DStorageAccount
+    listKind: M4DStorageAccountList
+    plural: m4dstorageaccounts
+    singular: m4dstorageaccount
+  scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      description: M4DStorageAccount defines a storage account used for copying data
+        It contains endpoint, region and a reference to the credentials a Owner of
+        the asset is responsible to store the credentials
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          description: M4DStorageAccountSpec defines the desired state of M4DStorageAccount
+          properties:
+            endpoint:
+              description: Endpoint
+              type: string
+            region:
+              description: Region
+              type: string
+            secretRef:
+              description: Secret including credentials for COS bucket
```

Comment:
We are creating Dataset CRDs with type: "COS" in the spec, so I assume it works only with COS. Especially, since they support only cos_hmac credentials.

Reply:
any S3 object store